salt/users/init.sls

```diff
@@ -4,7 +4,7 @@ include:
 {% for user_name, user_config in salt["pillar.get"]("users", {}).items() %}
 {% set admin = user_config.get("admin", false) %}
 {% set access = {} %}
-{% for pat, data in user_config.get("access", {}).items() if salt["match.compound"](salt["pillar.get"]("roles:" + pat)) %}  # " Syntax fix
+{% for pat, data in user_config.get("access", {}).items() if salt["match.compound"](salt["pillar.get"]("roles:" + pat + ":pattern")) %}  # " Syntax fix
```

Comment:
Salt allows for accessing nested keys with a `:` syntax, see https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.pillar.html#salt.modules.pillar.get.

This updates the `pillar.get` call to access the sub-key we need.

In an explicit example,

going from:

```
roles:
  salt: salt.nyc1.psf.io
```

```
roles:
  salt:
    pattern: salt.nyc1.psf.io
```

We need to go from accessing `roles:salt` to `roles:salt:pattern` which is why the `+ ":pattern"` is added to the string being queried for.